### DISCUSSION TITLE
Is manual istallation of Cloudflared possible?

### DISCUSSION THREAD

**Discussion by dentex** [homeassistant-apps/app-cloudflared#663] [Q&A] [answered]
Is it possible to manually install the Cloudflared add-on, I mean piece by piece, on a "core" HASS installation?
I'm still not very familiar with Home Assistant, but I can find my way into Linux systems.
I find difficult to switch to HASS OS or supervised installation.
My problem is that I have a tailored Debian 12 install on a Raspberry Pi 4 that I customized for my needs (i3 desktop with a bunch of scripts, multimedia stuff etc) where I run HASS core.

Thanks

**Comment by brenner-tobias** [comment, marked as answer]
Hi. You cannot install the add-on if you cannot use the add-on store in your installation.
What you can easily do though, is to install Cloudflared directly on your OS, by using the fitting binaries or using docker (see [downloads page](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/downloads/)).
Once you have this running, I suggest to follow our [explanation for a remote managed tunnel ](https://github.com/brenner-tobias/addon-cloudflared/wiki/How-tos#how-to-configure-remote-tunnels) and simply take the token not to the add-on but to your own run command (see [explanation from Cloudflared](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/get-started/create-remote-tunnel/))

**Comment by dentex** [comment]
Wonderful! thanks for the info.

**Comment by dentex** [comment]
I think I still have to adjust something... If I may ask for help, since I'm not using your add-on, this is what I've done until now:

- I set up the tunnel locally (it seems better: no credit card - no "zero trust" stuff)
- Prerequisites are OK: I have added a domain I bought at Ionos and changed its nameservers to Cloudflare
- I created a tunnel with url: http://localhost:8123
- I routed the tunnel to my domain
- Initially I forgot to "run" it, and got an error from Cloudflare (_Error 1033 - Argo Tunnel error_)
- Then, with the tunnel up and running, I got the "400: Bad Request" message in the browser. 

if I understand correctly, this is Home Assistant blocking requests from reverse proxies. Since I'm not running your add-on, do you know what I have to put under "trusted_proxies" (in HA `configuration.yaml`) instead of the docker network?

Thanks

**Reply by brenner-tobias** [reply]
Sure, in this case, the request from the "proxy" comes from your machine, under which you run cloudflared. So you have to enter the IP of the debian OS in there.

**Reply by dentex** [reply]
Yes, it makes sense.
I added the machine's local address to the config, so I have:

```
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 192.168.X.X
```

... but after restarting HA, I still have the bad request error    :(

**Reply by dentex** [reply]
Finally!

```
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 127.0.0.1
    - ::1
```

This was it.

THANKS for all the help :)

**Reply by punithrudrappa** [reply]
> Finally!
> 
> ```
> http:
>   use_x_forwarded_for: true
>   trusted_proxies:
>     - 127.0.0.1
>     - ::1
> ```
> 
> This was it.
> 
> THANKS for all the help :)

This worked for me too. Thanks for documenting it.
